packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js

```diff
@@ -0,0 +1,26 @@
+import { get } from 'lodash';
+
+const createPossibleMainFieldsForModelsAndComponents = array => {
+  return array.reduce((acc, current) => {
+    const attributes = get(current, ['schema', 'attributes'], {});
+    const possibleMainFields = Object.keys(attributes).filter(attr => {
+      return ![
+        'boolean',
+        'component',
+        'dynamiczone',
+        'json',
+        'media',
+        'password',
+        'relation',
+        'text',
+        'richtext',
+      ].includes(get(attributes, [attr, 'type'], ''));
+    });
+
+    acc[current.uid] = possibleMainFields;
+
+    return acc;
+  }, {});
+};
+
+export default createPossibleMainFieldsForModelsAndComponents;
```

Comment:
Some test for this function too.

Reply:
The tests will come in another PR after we are finished removing the saga dependency.

Reply:
Test have been added.